source/agora/consensus/EnrollmentManager.d

```diff
@@ -543,8 +543,9 @@ public class EnrollmentManager
         foreach (const ref key; keys)
         {
             const preimage = this.validator_set.getPreimageAt(key, height);
-            // TODO: The preimage should not have the .init value. So, checking
-            // that is needed.
+            // this should not happen. validators which didn't reveal the
+            // preimage should not be in the active validator set 'keys'.
+            assert(preimage != PreImageInfo.init);
```

Comment:
Looks like  the second commit should come first, so the assertion doesn't need to be deleted in the first place ?

Reply:
Yeah, It does not. I just wanted to make smaller PRs. I can merge the two commits.